assests/javascript/main.js

```diff
@@ -214,18 +212,55 @@ window.addEventListener('DOMContentLoaded', () => {
 
 /* Form validation */
 const contactForm = document.getElementById('contact-form');
-contactForm.addEventListener("submit", (event)=>{
-  let emailInput = contactForm.elements['email'];
-  let errorMsg =  document.getElementById('errorMsg');
+contactForm.addEventListener('submit', (event) => {
+  const emailInput = contactForm.elements.email;
+  const errorMsg = document.getElementById('errorMsg');
 
   /* check if the email is all in lower case or not */
-	if (emailInput.value !== emailInput.value.toLowerCase()) {
+  if (emailInput.value !== emailInput.value.toLowerCase()) {
     event.preventDefault();
     errorMsg.classList.add('errorMessage');
-    errorMsg.innerText = "Please enter the email address in lowercase!";
-  }
-  else {
+    errorMsg.innerText = 'Please enter the email address in lowercase!';
+  } else {
     errorMsg.classList.remove('errorMessage');
-    contactForm.submit()
+    contactForm.submit();
   }
 });
+
+/* Select input elements to capture data */
+const nameInput = document.getElementById('name');
+const emailInput = document.getElementById('email');
+const messageInput = document.getElementById('message');
+
+// Retrive data in localStorage
+function loadLocalStorage() {
+  const contactData = JSON.parse(window.localStorage.getItem('contactData'));
+  if (contactData) {
+    nameInput.value = contactData.name;
+    emailInput.value = contactData.email;
+    messageInput.value = contactData.message;
+  }
+}
+
+// Save in the local storage onchnage of the windowa */
+function onFormChange(event) {
+  let contactData = JSON.parse(window.localStorage.getItem('contactData'));
+
+  if (!contactData) {
+    contactData = {};
+  }
+
+  const key = event.target.name;
+  contactData[key] = event.target.value;
+  contactData = JSON.stringify(contactData);
+  window.localStorage.setItem('contactData', contactData);
+}
+
+nameInput.addEventListener('change', onFormChange);
+emailInput.addEventListener('change', onFormChange);
+messageInput.addEventListener('change', onFormChange);
+
+// Load data in DOM on windows load
+window.onload = () => {
+  loadLocalStorage();
+};
```

Comment:
- [ ] When filling the form I noticed these things:
- No matter what I put under name when I reload I get undefined under it
- The email and message get saved and when I reload I can see them. But if I leave them empty and reload I get the undefined word assigned to them(_Note: this happens only when I first load the page example: if I write something under email or message reload then delete and reload again, I will get them as empty, but if for a first time I leave them empty and reload I get undefined under word on their input_)

![image](https://user-images.githubusercontent.com/62705947/203961941-4e174a12-a688-4b5b-ac27-c7d8655e46e2.png)